.github/workflows/jekyll.yml

```diff
@@ -0,0 +1,54 @@
+name: Jekyll site CI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-ruby@v1
+      with:
+        ruby-version: '2.x'
+    - name: Install dependencies
+      run: bundle install
+    - name: Build site and feed
+      run: |
+        # have site.url produce a valid value
+        sed -i -e '/^collections:/i \
+        url: https://rustacean-station.org' \
+               _config.yml
+        bundle exec jekyll build
+    - name: Check dates
+      run: |
+        for date in $(grep -h date _episodes/*.md | sed 's/^date: //'); do
+          date "--date=$date";
+        done
+    - name: Check file sizes
+      run: |
+        for episode in _episodes/*.md; do
+          length=$(grep 'length:' "$episode" | awk '{print $2}' | sed 's/"//g');
+          file=$(grep 'file:' "$episode" | sed -e 's/^file: //' -e 's/"//g');
+          size=$(curl --head -s "$file" | grep content-length | awk '{print $2}' | sed 's/\r//')
```

Comment:
Are you worried about having every new commit to every new PR download every single episode? I don't suppose there's an easy way to have `for episode in _episodes/*.md` be something more like `for episode in [episodes whose .md files have actually changed]`?

Reply:
Keep in mind that it's just doing a `HEAD` request, not downloading the entire file! I don't think there's a way of checking only files that have changed with the GitHub Action API (at least from a cursory look at the [docs](https://help.github.com/en/actions)), though we could of course do something like only look at files made in the last, say, month. Then we could also actually download the file and check that the `duration` field is set correctly. What's your preferred heuristic here?

Reply:
Ah yes, I missed the head part. :P That should be fine.